csrc/rsa_cipher.cpp

```diff
@@ -61,15 +62,17 @@ JNIEXPORT jint JNICALL Java_com_amazon_corretto_crypto_provider_RsaCipher_cipher
             throw_java_ex(EX_NPE, "Null output array");
         }
 
-        EvpKeyContext *ctx = reinterpret_cast<EvpKeyContext *>(ctxHandle);
-
-        EVP_PKEY_CTX *keyCtx = ctx->getKeyCtx();
-        if (keyCtx == nullptr) {
-            keyCtx = ctx->setKeyCtx(EVP_PKEY_CTX_new(ctx->getKey(), /*engine*/nullptr));
-        }
+        EVP_PKEY* key = reinterpret_cast<EVP_PKEY*>(ctxHandle);
+        // It is unclear if allocating a new EVP_PKEY_CTX for each operation is expensive.
+        // If profiling shows that this is a significant cost when in a hot-path we should figure out a way to
```

Comment:
out of curiosity, what profiling tools do we use for the native portions of ACCP? i.e. what tool would we use to determine whether or not this is a hotspot?

Reply:
A combination of any benchmark tools we have and [async-profiler](https://github.com/jvm-profiling-tools/async-profiler/) and [perf](https://en.wikipedia.org/wiki/Perf_%28Linux%29).